core/src/main/java/software/amazon/awssdk/core/util/DefaultSdkAutoConstructAwareList.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.core.util;
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+/**
+ * Default implementation of {@link SdkAutoConstructAwareList}.
+ *
+ * @param <T> The element type.
+ */
+public class DefaultSdkAutoConstructAwareList<T> extends ArrayList<T> implements SdkAutoConstructAwareList<T> {
+    private final boolean isAutoConstructed;
+
+    public DefaultSdkAutoConstructAwareList() {
+        super();
+        isAutoConstructed = true;
+    }
+
+    public DefaultSdkAutoConstructAwareList(Collection<? extends T> other) {
```

Comment:
Do we need these other constructors or can we just use the list implementation they pass in?

Reply:
I'd prefer to keep this since we make use of the copy constructor in in the model copier classes.

Reply:
Removed per the above comment.